src/coreclr/utilcode/pedecoder.cpp

```diff
@@ -776,8 +775,10 @@ IMAGE_SECTION_HEADER *PEDecoder::RvaToSection(RVA rva) const
 
     while (section < sectionEnd)
     {
-        if (rva < (VAL32(section->VirtualAddress)
-                   + AlignUp((UINT)VAL32(section->Misc.VirtualSize), (UINT)VAL32(FindNTHeaders()->OptionalHeader.SectionAlignment))))
+        // The RVA should be within a section's virtual address range. 
+        // On flat images (!IsMapped()), the RVA should also be within the section's raw data range.
+        if (rva < (VAL32(section->VirtualAddress) + VAL32(section->Misc.VirtualSize)) &&
+            (IsMapped() || rva < (VAL32(section->VirtualAddress) + VAL32(section->SizeOfRawData))))
```

Comment:
Could you please also delete "/ !!! check for case where rva is in padded portion of segment" comment later in the file?